ath79: fix led_power GPIO for D-Link DIR-842

s-2 opened the pull request:
The device has a total of 8 LEDs, 5 of which are controlled by the switch
(LAN 1-4, WAN). Only power, wifi and wps are controlled by the SoC.

 * led_power is on GPIO 5 (not 15), boot flashing sequence is now visible
 * remove led 'internet', since it is only connected to the switch
 * remove ucidef_set_led_switch for WAN from 01_leds, as it has no effect
 * replace gpio-hog on GPIO 11 with phy0 reset-gpios (for external switch)
 * increase spi-max-frequency to 50 MHz

Tested on revisions C1 and C3.

Signed-off-by: Sebastian Schaper <openwrt@sebastianschaper.net>

s-2 commented:
GPIO 11 `export-gpio` on DIR-859 should probably be changed to `phy0 reset-gpios` as well, however I don't own that device to verify.

adrianschmutzler commented:
Please split this into three separate commits (within this PR):
- LED changes
- frequency change
- gpio-hop/reset-gpio changes

s-2 committed:
ath79: fix led_power GPIO for D-Link DIR-842

The device has a total of 8 LEDs, 5 of which are controlled by the switch
(LAN 1-4, WAN). Only power, wifi and wps are controlled by the SoC.

 * led_power is on GPIO 5 (not 15), boot flashing sequence is now visible
 * remove led 'internet', since it is only connected to the switch
 * remove ucidef_set_led_switch for WAN from 01_leds, as it has no effect

Tested on revisions C1 and C3.

Signed-off-by: Sebastian Schaper <openwrt@sebastianschaper.net>

s-2 committed:
ath79: define switch reset-gpios for D-Link DIR-842

GPIO 11 needs to be pulled high for the external gigabit switch to work,
this is currently solved via gpio-hog. Replace with phy0 reset-gpios.

Tested on revisions C1 and C3. Reset button is still working for reboot,
to enter failsafe, and to enter bootloader http recovery.

Signed-off-by: Sebastian Schaper <openwrt@sebastianschaper.net>

s-2 committed:
ath79: increase spi clock for D-Link DIR-842

AHB is 258 MHz for this device (CPU_PLL / 3), but there is no difference
between 64 MHz and 50 MHz for spi-max-frequency, thus increase to 50 MHz.

Tested on revisions C1 and C3.

Signed-off-by: Sebastian Schaper <openwrt@sebastianschaper.net>

adrianschmutzler commented:
@s-2 Thanks, I've pulled this to my staging tree.

@hackpascal Can you check/test whether the changes here will apply for the DIR-859-A1 as well?
